### PR TITLE
Added libraries and dependencies

### PR DESCRIPTION
**Pull request by jeffh1996**
Regarding issue #42 . I have added the additional libraries lib32ncurses5, libncurses5:i386, and software-properties-common.

Also, the stlink, CubeMX2MakeFile, and gcc4mbed versions now come from the team's repository.

### CODE REVIEW

**Comment by enochtsang** [comment]
@jeffh1996 reference the issue number next time

**Comment by bill-luu** [comment]
Thanks Jeff! I assigned you issues on the stlink and gcc4mbed repos, can you take care of those too and then we'll test this script to see if it works.

File: STM32F4Discovery-Resources/install.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ apt-get -y install build-essential git libsane:i386 ia32-libs-multiarch autoconf
 # Install STLink
 if ! type "st-flash" > /dev/null; then
     echo "*** Installing ST Link"
-    git clone https://github.com/UCSolarCarTeam/stlink.git /opt/stlink --depth 1
```
Comment (bill-luu) [inline STM32F4Discovery-Resources/install.sh]: How come this was changed back to the `texane` repo instead of the `UCSolarCarTeam` one?

**Comment by jeffh1996** [comment]
Sorry about that. Fixed that line. Also, I have no idea why travis is failing.

**Comment by bill-luu** [comment]
Having Louis test this

**Comment by louisj381** [comment]
Through testing, found that script ran properly, however when gcc-arm-none-eabi is added to .profile, it may not be updating through path. When running the make command I was getting an error that "gcc-arm... is not found". When I tried another terminal I had the same error. However, when I restarted my computer and ran the command it was successful.

**Pull request merged**